airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -171,6 +170,11 @@ private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterIn
         workflowState.setResetWithScheduling(true);
       }
 
+      // Clean the job state by failing any jobs for this connection that are currently non-terminal.
+      // This catches cases where the temporal workflow was terminated and restarted while a job was
+      // actively running, leaving that job in an orphaned and non-terminal state.
+      ensureCleanJobState(connectionUpdaterInput);
```

Comment:
Had to move this down below the statements that transfer values from the `connectionUpdaterInput` into the `workflowState`, so that if the `ensureCleanJobState` activity fails, the resetConnection flag is still set as desired on the workflowState, as that is used in the workflow retrying logic below.